File: src/cli-format.ts
import chalk from 'chalk';
import { McpListing, SafeUser } from './domain/models.js';

export function printSuccess(message: string): void {
  console.log(chalk.green(`✔ ${message}`));
}

export function printError(message: string): void {
  console.error(chalk.red(`✖ ${message}`));
}

export function printInfo(message: string): void {
  console.log(chalk.cyan(`ℹ ${message}`));
}

export function printUser(user: SafeUser): void {
  console.log(chalk.bold('当前用户'));
  console.log(`- ID: ${user.id}`);
  console.log(`- 邮箱: ${user.email}`);
  console.log(`- 显示名: ${user.displayName}`);
}

export function printListings(listings: McpListing[]): void {
  if (listings.length === 0) {
    printInfo('当前没有符合条件的 MCP 记录');
    return;
  }

  for (const listing of listings) {
    console.log(chalk.bold(`
${listing.title} (${listing.id})`));
    console.log(`  简介: ${listing.summary}`);
    console.log(`  版本: ${listing.version}`);
    console.log(`  传输: ${listing.transport}`);
    console.log(`  认证: ${listing.authType}`);
    console.log(`  来源: ${listing.sourceType === 'endpoint' ? listing.endpointUrl : `${listing.packageRegistry}:${listing.packageName}`}`);
    console.log(`  标签: ${listing.tags.join(', ') || '无'}`);
    console.log(`  任务: ${listing.taskTypes.join(', ') || '无'}`);
    console.log(`  创建时间: ${listing.createdAt}`);
  }
}

File: src/cli_format.py
import sys


def print_success(message: str) -> None:
    print(f"✔ {message}")


def print_error(message: str) -> None:
    print(f"✖ {message}", file=sys.stderr)


def print_info(message: str) -> None:
    print(f"ℹ {message}")


def print_user(user) -> None:
    print("当前用户")
    print(f"- ID: {user.id}")
    print(f"- 邮箱: {user.email}")
    print(f"- 显示名: {user.display_name}")


def print_listings(listings) -> None:
    if not listings:
        print_info("当前没有符合条件的 MCP 记录")
        return

    for listing in listings:
        print(f"\n{listing.title} ({listing.id})")
        print(f"  简介: {listing.summary}")
        print(f"  版本: {listing.version}")
        print(f"  传输: {listing.transport}")
        print(f"  认证: {listing.auth_type}")
        if listing.source_type == "endpoint":
            source = listing.endpoint_url
        else:
            source = f"{listing.package_registry}:{listing.package_name}"
        print(f"  来源: {source}")
        print(f"  标签: {', '.join(listing.tags) if listing.tags else '无'}")
        print(f"  任务: {', '.join(listing.task_types) if listing.task_types else '无'}")
        print(f"  创建时间: {listing.created_at}")
